rspace/src/main/scala/coop/rchain/rspace/package.scala

```diff
@@ -103,56 +103,57 @@ package object rspace {
                           channels: Seq[C],
                           patterns: Seq[P],
                           continuation: K,
-                          persist: Boolean)(implicit m: Match[P, A]): Option[(K, Seq[A])] = {
-    if (channels.length =!= patterns.length) {
-      val msg = "channels.length must equal patterns.length"
-      logger.error(msg)
-      throw new IllegalArgumentException(msg)
-    }
-    store.withTxn(store.createTxnWrite()) { txn =>
-      logger.debug(s"""|consume: searching for data matching <patterns: $patterns> 
-                       |at <channels: $channels>""".stripMargin.replace('\n', ' '))
-      /*
-       * Here, we create a cache of the data at each channel as `channelToIndexedData`
-       * which is used for finding matches.  When a speculative match is found, we can
-       * remove the matching datum from the remaining data candidates in the cache.
-       *
-       * Put another way, this allows us to speculatively remove matching data without
-       * affecting the actual store contents.
-       */
+                          persist: Boolean)(implicit m: Match[P, A]): Option[(K, Seq[A])] =
+    store.eventsCounter.registerConsume {
+      if (channels.length =!= patterns.length) {
```

Comment:
Let's move this check outside of the `registerConsume`.

Reply:
This may lead to a non-intuitive behavior, since call will exist in unit-tests or real code, but not counted. Is it ok?

Reply:
Hmmm.

Yr right, this is the most consistent way of handling errors for now (considering other exceptions could happen inside the body).

Put alternately, we aren't counting successful consumes, just calls to consume.

Okay, I agree.